### PR TITLE
Update row chart measuring function

### LINKED ISSUE #32044: Row charts do not render when aggregating
### Describe the bug

When you are aggregating by a custom column, the row chart fails to render:

image

### To Reproduce

1. Go to Sample Data 
2. Create a New GUI question on the Products table
3.  Summarize Count by Category
4. Hit Visualize
5. Change the Visualization to a Row chart

### Expected behavior

Rows should appear

### Logs

_No response_

### Information about your Metabase installation

```JSON
- On Chrome, Windows
- I'm seeing this on the build from  2023-06-30
```

### Severity

Upgrade blocker.  Previously working questions stopped working

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by iethree**
closes https://github.com/metabase/metabase/issues/32044, except that the issue is much bigger than the issue stated, this broke all row charts 😳 

### Description

#31722 changed the way we measure text in several visualizations, and while it updated the types in RowChart, it didn't actually handle the fact that we are now passing it a function that returns width and height, while the RowChart component expects a function that only returns a width value.  This broke all row chart visualizations. Because we passed in a custom `measureText` function in the unit tests, unit tests did not catch it.

> 🤔 _I wonder if we should have unit tests at the `Visualization` level for all our visualizations_

This simply transforms the function that is actually passed into a form that RowChart expects.

### How to verify

make any question and make it into a rowChart

### Demo

Before | After
--- | ---
Screen Shot 2023-06-30 at 12 20 52 PM |  Screen Shot 2023-06-30 at 12 20 41 PM

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/test/scenarios/visualizations/rows.cy.spec.js [hunk 1]
```diff
@@ -43,4 +43,41 @@ describe("scenarios > visualizations > rows", () => {
       },
     );
   });
+
+  it(`should display a row chart`, () => {
```
Comment (alxnddr) [inline e2e/test/scenarios/visualizations/rows.cy.spec.js:47]: 🙏🏻

**Review by alxnddr — APPROVED** [review]

File: frontend/src/metabase/visualizations/shared/components/RowChart/RowChart.tsx [hunk 2]
```diff
@@ -89,7 +92,7 @@ export const RowChart = <TDatum,>({
 
   xScaleType = "linear",
 
-  measureText,
+  measureText: measureTextSize,
```
Comment (WiNloSt) [inline frontend/src/metabase/visualizations/shared/components/RowChart/RowChart.tsx:95]: What do you think about keeping `measureText` as before i.e. only measures width? And create a new function called `measureTextDimension` that returns `width` and `height` for https://github.com/metabase/metabase/blob/3b08aaf4cd215b2426020131d703f369f1a2b080/frontend/src/metabase/visualizations/components/ScalarValue/utils.ts#L28-L32

This way we don't have different APIs for `measureText` which could create confusion and cause name conflicts just like this.

**Review by WiNloSt — COMMENTED** [review]
Both viz and static viz work wonderfully 🎉 . I've commented regarding the use of `measureText` variable. Let me know what you think.

File: frontend/src/metabase/static-viz/components/RowChart/RowChart.tsx [hunk 3]
```diff
@@ -51,16 +51,19 @@ interface StaticRowChartProps {
...
+const staticTextMeasurer: TextMeasurer = (text: string, style: FontStyle) => {
+  const textWidth = measureText(
     text,
     parseInt(style.size.toString(), 10),
     style.weight ? parseInt(style.weight.toString(), 10) : 400,
   );
 
+  return {
+    width: textWidth,
+    height: -1, // this will be ignored by RowChart
```
Comment (kamilmielnik) [inline frontend/src/metabase/static-viz/components/RowChart/RowChart.tsx:63]: This is a bit confusing. If `height` is not used, then previous interface made more sense.

How about we apply [this suggestion](https://github.com/metabase/metabase/pull/32057/files#r1250027757) but improve the naming, i.e. `measureTextWidth: TextWidthMeasurer` + `measureTextDimensions: TextDimensionsMeasurer`? 

`measureText` would be no more.

**Review by kamilmielnik — APPROVED** [review]
:pray:

**Comment by iethree** [comment]
@kamilmielnik @WiNloSt I'm gonna merge this as-is for now to make sure it all gets backported in time for RC3 🤞 . I'll open up a follow-up PR to clean up the naming conventions.

**Pull request merged**